webpack/assets/javascripts/react_app/components/users/PersonalAccessTokens/PersonalAccessTokenModal.js

```diff
@@ -0,0 +1,289 @@
+  };
+
+  const validateTimeChange = newTime => {
+    if (!newTime.length) setIsTimeValid(true);
+    else {
+      const splitme = newTime.split(':');
+      if (
+        !(
+          splitme[0].length === 2 &&
+          splitme[1]?.length === 2 &&
+          splitme[2]?.length === 2
+        )
+      )
+        setIsTimeValid(false);
+      else if (!Number.isNaN(new Date(`${date} ${newTime}`).getTime()))
+        setIsTimeValid(true);
+      else setIsTimeValid(false);
+    }
+    setTime(newTime);
+  };
+
+  const tokens = useSelector(state => selectTokens(state));
+  const isNameDuplicate = () => tokens.find(obj => obj.name === name);
+  const isNameEmpty = () => name.length === 0;
+
+  const nameHelperText = () => {
+    if (showNameErrors) {
+      if (isNameEmpty()) {
+        return __('Fill out the name');
+      } else if (isNameDuplicate() !== undefined) {
+        return __('Name has already been taken');
+      }
+    }
+    return '';
+  };
+
+  const isDateTimeInFuture = () => {
+    if (date.length !== 0) {
+      const chosenDate = new Date(`${date} ${time}`);
+      const currentDate = new Date();
+      if (chosenDate.getTime() <= currentDate.getTime()) {
+        return false;
+      }
+      return true;
+    }
+    return true;
+  };
+
+  const handleSubmit = () => {
+    if (isDateTimeInFuture() && isDateValid && isTimeValid) {
+      dispatch(
+        APIActions.post({
+          key: PERSONAL_ACCESS_TOKEN_FORM_SUBMITTED,
+          url,
+          params: { name, expires_at: `${date} ${time}`, controller },
+          handleSuccess: ({ data }) => {
+            handleModalToggle();
+            dispatch({
+              type: PERSONAL_ACCESS_TOKEN_FORM_SUBMITTED,
+              payload: { item: 'personal_access_token', data },
+            });
+          },
+          successToast: () =>
+            __('Personal Access Token was successfully created.'),
+          errorToast: ({ response }) =>
+            response?.data?.error?.message ||
+            response?.message ||
+            response?.statusText,
+        })
+      );
+    }
+  };
+
+  return (
+    <>
+      <Button
+        aria-label="add-token-button"
+        variant="primary"
+        isSmall
+        onClick={handleModalToggle}
+      >
+        {__('Add Personal Access Token')}
+      </Button>
+      <Modal
+        id="new-token-modal"
+        className="token-modal"
+        variant={ModalVariant.small}
+        title={__('Create Personal Access Token')}
+        isOpen={isModalOpen}
+        onClose={handleModalToggle}
+        actions={[
+          <Button
+            aria-label="confirm-button"
+            key="confirm"
+            variant="primary"
+            onClick={() => {
+              handleSubmit();
+              setShowNameErrors(true);
+            }}
+            isDisabled={isSubmitting}
+          >
+            Confirm
+          </Button>,
+          <Button key="cancel" variant="link" onClick={handleModalToggle}>
```

Comment:
Should also be disabled with `isSubmitting` as clicking on it will not cancel the submit